get_http_test.go

```diff
@@ -181,28 +187,39 @@ func TestHttpGetter_resume(t *testing.T) {
 	if err != nil {
 		t.Fatalf("create: %v", err)
 	}
-	f.Write(load[:downloadFrom])
-	f.Close()
+	if n, err := f.Write(load[:downloadFrom]); n != downloadFrom || err != nil {
+		t.Fatalf("partia file write failed: %d, %s", n, err)
+	}
+	if err := f.Close(); err != nil {
+		t.Fatalf("close failed: %s", err)
+	}
 
-	var u url.URL
-	u.Scheme = "http"
-	u.Host = ln.Addr().String()
-	u.Path = "/range"
+	u := url.URL{
+		Scheme:   "http",
+		Host:     ln.Addr().String(),
+		Path:     "/range",
+		RawQuery: "checksum=" + checksum,
+	}
+	t.Logf("url: %s", u.String())
 
-	// Get it!
-	if err := g.GetFile(dst, &u); err != nil {
-		t.Fatalf("should not error: %v", err)
+	// Finish getting it!
+	if err := GetFile(dst, u.String()); err != nil {
+		t.Errorf("finishing download should not error: %v", err)
 	}
 
 	b, err := ioutil.ReadFile(dst)
 	if err != nil {
-		t.Fatalf("should not error: %v", err)
+		t.Fatalf("readfile failed: %v", err)
 	}
 
 	if string(b) != string(load) {
-		t.Fatalf("file differs: got:\n%s\n expected:\n%s\n", string(b), string(load))
+		t.Errorf("file differs: got:\n%s\n expected:\n%s\n", string(b), string(load))
 	}
 
+	// Get it again
+	if err := GetFile(dst, u.String()); err != nil {
+		t.Errorf("should not error: %v", err)
```

Comment:
As with the other `Errorf` messages here.

As a general rule, `Fatalf` is usually good as if basically indicates that the test case is now "tainted" in a sense in that there's no way we can continue reliably. `Errorf` is fine if we have multiple errors, but how can we be sure that those errors are valid errors in the event of a failure up the line?

Reply:
See https://github.com/hashicorp/go-getter/pull/163#discussion_r254781399 also